src/ripple/app/ledger/LedgerEntrySet.cpp

```diff
@@ -1406,6 +1422,40 @@ TER LedgerEntrySet::trustDelete (
     return terResult;
 }
 
+void LedgerEntrySet::enableDeferredCredits (bool enable)
+{
+    if (!enable)
+    {
+        mDeferredCredits.reset ();
+        return;
+    }
+
+    if (!mDeferredCredits)
+        mDeferredCredits.emplace ();
+}
```

Comment:
That makes sense. I didn't notice the implication in payments, but definitely not appropriate to add the assert as such.